sycl/doc/extensions/proposed/sycl_ext_intel_fpga_kernel_interface_properties.asciidoc

```diff
@@ -0,0 +1,296 @@
+inline constexpr pipelined_key::value_t<pipeline_directive,
+  initiation_interval> pipelined;
+
+} // namespace sycl::ext::oneapi::experimental
+```
+
+|===
+|Property|Description
+|`streaming_interface`
+|The `streaming_interface` property adds the requirement that the kernel must
+ have dedicated ports for input / output signals. This applies for both
+ control, and kernel argument data signals. The following values are supported:
+
+ * `accept_downstream_stall`: Directs the compiler to generate a kernel
+    interface that can accept a back-pressure signal.
+
+ * `remove_downstream_stall`: Directs the compiler to generate a kernel
+    interface that does not accept a back-pressure signal.
+
+ If the `streaming_interface` property is not specified, the default behavior is
+ equivalent to one of the values listed above, but the choice is implementation
+ defined.
+
+ The following properties have been provided for convenience:
+ `streaming_interface_accept_downstream_stall`,
+ `streaming_interface_remove_downstream_stall`.
+
+|`register_map_interface`
+|The `register_map_interface` property adds the requirement that the kernel must
+ have its input / output control and kernel argument data signals placed in a
+ shared Control and Status Register (CSR) map. The following values are
+ supported:
+
+ * `wait_for_done_write`: Directs the compiler to generate logic that
+    back-pressures the kernel until the kernel is notified that its completion
+    has been detected. The kernel will be notified when the register it writes
+    its completion signal to, is set to 1.
+
+ * `do_not_wait_for_done_write`: Directs the compiler to not generate logic that
+    would back-pressure the kernel until the kernel is notified of its
+    completion being detected.
+ 
+ If the `register_map_interface` property is not specified, the default behavior
+ is equivalent to one of the values listed above, but the choice is
+ implementation defined.
+
+ The following properties have been provided for convenience:
+ `register_map_interface_wait_for_done_write`,
+ `register_map_interface_do_not_wait_for_done_write`.
+
+|`pipelined`
+|Two sets of values are accepted as property parameters - a boolean, and another
+ argument `initiation_interval`.
+
+ When the boolean parameter is set to `true`, the property directs the compiler
+ to pipeline calls to the kernel such that multiple invocations of the kernel
+ can be in flight simultaneously. When the boolean parameter is set to `false`,
+ the compiler will not pipeline kernel invocations.
+
+ The second argument specifies how many cycles can exist between successive
+ invocations of the kernel:
+
+ * `true, initiation_interval<N>`: The compiler will pipeline multiple kernel
+   invocations such that an invocation is launched every N cycles.
+ 
+ If the `initiation_interval` argument is not specified, the compiler will
+ attempt to launch the kernel invocations as fast as it can.
```

Comment:
I propose we leave this as an implementation detail.  As written it implies that an implementation must aggressively pursue II=1 even if that doesn't make sense (e.g. because the kernel contains a loop with II=1 with high trip count which will be the bottleneck anyways).

Reply:
Updated to mark this as implementation defined.